softwareComponents/configuration/include/configuration/serialization.hpp

```diff
@@ -89,19 +106,21 @@ namespace rofi::configuration::serialization {
             j = component;
     }
 
-    inline nlohmann::json toJSON( const UniversalModule& m ) {
+    template< typename Callback >
+    inline nlohmann::json moduleToJSON( const UniversalModule& m, Callback& attrCb ) {
         using namespace nlohmann;
         json j;
         j[ "type"  ] = "universal";
         j[ "alpha" ] = m.getAlpha().deg();
         j[ "beta"  ] = m.getBeta().deg();
         j[ "gamma" ] = m.getGamma().deg();
 
+        addAttributes( j, attrCb, m );
         return { { std::to_string( m.getId() ), j } };
     }
 
     template<>
-    inline UniversalModule fromJSON( const nlohmann::json& j, ModuleId id ) {
+    inline UniversalModule moduleFromJSON( const nlohmann::json& j, ModuleId id ) {
```

Comment:
Consider adding the specification type explicitly `inline UniversalModule moduleFromJSON<UniversalModule>(`. I'm not 100% sure it's much better, since you repeat yourself, but it makes clear for me that it's a specialization.